src/keys.rs

```diff
@@ -133,11 +137,25 @@ impl PartialEq for SpendValidatingKey {
     }
 }
 
+impl Eq for SpendValidatingKey {}
+
 impl SpendValidatingKey {
     /// Randomizes this spend validating key with the given `randomizer`.
     pub fn randomize(&self, randomizer: &pallas::Scalar) -> redpallas::VerificationKey<SpendAuth> {
         self.0.randomize(randomizer)
     }
+
+    /// Converts this spend validating key to its serialized form.
+    pub(crate) fn to_bytes(&self) -> [u8; 32] {
+        <[u8; 32]>::from(&self.0)
+    }
+
+    pub(crate) fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        <[u8; 32]>::try_from(&bytes[..32])
```

Comment:
Why does this take the first 32 bytes? Shouldn't it fail if the length isn't exactly 32 bytes?